### ISSUE TITLE
-Werror causes failures with Kotlin 1.4.0

### ISSUE THREAD

**Issue by DanySK** [mockk/mockk#483]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### Expected Behavior

Mockk works with Kotlin 1.4.0 as well as it did with Kotlin 1.3.x

### Current Behavior

Well, it works, but if you use `-Werror` (and I do), then it causes the following issue:

```
> Task :02-kotlin-helloworld:compileTestKotlin FAILED
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    /home/danysk/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk8/1.4.0/e3765b66f0610afc92053ff1a93a87a544fca2b/kotlin-stdlib-jdk8-1.4.0.jar (version 1.4)
    /home/danysk/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk7/1.4.0/9cc187c3dfaf6e4001bdf962e3cdadff7690261b/kotlin-stdlib-jdk7-1.4.0.jar (version 1.4)
    /home/danysk/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.3.50/b499f22fd7c3e9c2e5b6c4005221fa47fc7f9a7a/kotlin-reflect-1.3.50.jar (version 1.3)
    /home/danysk/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.4.0/63e75298e93d4ae0b299bb869cf0c627196f8843/kotlin-stdlib-1.4.0.jar (version 1.4)
    /home/danysk/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.4.0/1c752cce0ead8d504ccc88a4fed6471fd83ab0dd/kotlin-stdlib-common-1.4.0.jar (version 1.4)
w: Consider providing an explicit dependency on kotlin-reflect 1.4 to prevent strange errors
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath
e: warnings found and -Werror specified
```

### Context

This is not really a bug of mockk itself. It's more of a Kotlin issue, related to [this](https://youtrack.jetbrains.com/issue/KT-41142).
Actually, this is a valid dependency management issue, as *there are* differently version kotlin packages in the classpath,
and it's no one fault.
This bug will resolve itself as soon as a new Mockk depending on Kotlin 1.4.x will be released (see #475).
I wanted to prepare this report mostly to be sure people can find it on the Internet when pasting the exception,
and find the workaround.

### Workaround

Applying the suggested workaround works: 
```kotlin
dependencies {
    implementation(kotlin("stdlib-jdk8"))
    testImplementation("io.mockk:mockk:_")
    testImplementation(kotlin("reflect"))
}
```

**Comment by DanySK** [comment]
Well this is still an issue I think

**Comment by Raibaz** [comment]
Thanks for submitting this, it should be fixed by 1.10.3, as it is compiled with kotlin 1.4.20.

**Comment by Raibaz** [comment]
Still an issue indeed, but it should be fixed by 1.11.0 which will feature the upgrade to 1.4.
